browser/ui/containers/containers_menu_model.h

```diff
@@ -0,0 +1,66 @@
+// Copyright (c) 2025 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#ifndef BRAVE_BROWSER_UI_CONTAINERS_CONTAINERS_MENU_MODEL_H_
+#define BRAVE_BROWSER_UI_CONTAINERS_CONTAINERS_MENU_MODEL_H_
+
+#include <limits>
+#include <vector>
+
+#include "base/memory/raw_ref.h"
+#include "brave/browser/ui/containers/container_model.h"
+#include "brave/components/containers/core/mojom/containers.mojom.h"
+#include "ui/menus/simple_menu_model.h"
+
+FORWARD_DECLARE_TEST(ContainersMenuModelUnitTest, ModelContainsAllContainers);
+
+class Browser;
+
+namespace containers {
+
+class ContainersMenuModel : public ui::SimpleMenuModel,
+                            public ui::SimpleMenuModel::Delegate {
+ public:
+  class Delegate {
+   public:
+    Delegate() = default;
+    Delegate(const Delegate&) = delete;
+    Delegate& operator=(const Delegate&) = delete;
+    virtual ~Delegate() = default;
+
+    virtual void OnContainerSelected(
+        const containers::mojom::ContainerPtr& container) = 0;
+    virtual base::flat_set<std::string_view> GetCurrentContainerIds() = 0;
```

Comment:
this should be a set of `std::string`. A delegate can't guarantee that these `std::string_view` will be alive after the call.